zerver/lib/users.py

```diff
@@ -13,14 +13,17 @@
 
 from zulip_bots.custom_exceptions import ConfigValidationError
 
+from unicodedata import category
+
 def check_full_name(full_name_raw: str) -> str:
     full_name = full_name_raw.strip()
     if len(full_name) > UserProfile.MAX_NAME_LENGTH:
         raise JsonableError(_("Name too long!"))
     if len(full_name) < UserProfile.MIN_NAME_LENGTH:
         raise JsonableError(_("Name too short!"))
-    if list(set(full_name).intersection(UserProfile.NAME_INVALID_CHARS)):
-        raise JsonableError(_("Invalid characters in name!"))
+    for character in full_name:
+        if (category(character)[0] == 'C') or (character in UserProfile.NAME_INVALID_CHARS):
```

Comment:
what's the `category(character)` doing here?

Reply:
When `category` is called on a character, a two character string is returned that describes the type of character it is in the unicode standard. The first letter is the only one checked here as the first one is a more general descriptor than the second. Any code that starts with `C` is a unicode control character of some kind.

For a more specific description [reference here](https://stackoverflow.com/a/19016117).